### PR TITLE
DBZ-625 Fix parsing of float without decimals

### PR DESCRIPTION
**Pull request by jpechane**
https://issues.jboss.org/browse/DBZ-625

### CODE REVIEW

File: debezium-connector-postgres/src/test/java/io/debezium/connector/postgresql/AbstractRecordsProducerTest.java [hunk 1]
```diff
@@ -147,6 +162,21 @@
...
     }
 
+    protected List<SchemaAndValueField> schemasAndValuesForNumericDecimalTypeWithoutDecimals() {
+        final Struct dvs = new Struct(VariableScaleDecimal.schema());
+        dvs.put("scale", 1).put("value", new BigDecimal("10.0").unscaledValue().toByteArray());
+        final Struct nvs = new Struct(VariableScaleDecimal.schema());
+        nvs.put("scale", 1).put("value", new BigDecimal("22.0").unscaledValue().toByteArray());
+        return Arrays.asList(
+                new SchemaAndValueField("d", Decimal.builder(2).optional().build(), new BigDecimal("1.00")),
+     // DBZ-351 new SchemaAndValueField("dzs", Decimal.builder(0).optional().build(), new BigDecimal("10")),
```
Comment (gunnarmorling) [inline debezium-connector-postgres/src/test/java/io/debezium/connector/postgresql/AbstractRecordsProducerTest.java]: Is this commented line a left-over (same below)?

File: debezium-core/src/main/java/io/debezium/document/ComparableValue.java [hunk 2]
```diff
@@ -203,6 +203,7 @@ public Float asFloat() {
             double raw = ((Double) value).doubleValue();
             if (isValidFloat(raw)) return Float.valueOf((float) raw);
         }
+        if (value instanceof Number) return ((Number)value).floatValue();
```
Comment (gunnarmorling) [inline debezium-core/src/main/java/io/debezium/document/ComparableValue.java]: `floatValue()` seems prone to precision loss, shouldn't we call `doubleValue()` instead?

File: debezium-connector-postgres/src/test/java/io/debezium/connector/postgresql/AbstractRecordsProducerTest.java (hunk 1 above)
Comment (jpechane) [reply debezium-connector-postgres/src/test/java/io/debezium/connector/postgresql/AbstractRecordsProducerTest.java]: No, please check the Jira issue, we still pass numeric as double from protobug decoder

File: debezium-core/src/main/java/io/debezium/document/ComparableValue.java (hunk 2 above)
Comment (jpechane) [reply debezium-core/src/main/java/io/debezium/document/ComparableValue.java]: Yes, but the result of the call should be `float` so it is probably unnecessary to call `doubleValue()` and then in next step to convert it to `float`.

Comment (gunnarmorling) [reply debezium-core/src/main/java/io/debezium/document/ComparableValue.java]: Ah yes. I missed that we're in `asFloat()` already :)

File: debezium-connector-postgres/src/test/java/io/debezium/connector/postgresql/AbstractRecordsProducerTest.java (hunk 1 above)
Comment (gunnarmorling) [reply debezium-connector-postgres/src/test/java/io/debezium/connector/postgresql/AbstractRecordsProducerTest.java]: Ah, ok. Just assigned that one to 0.8.0, it's about time this gets fixed.

**Comment by gunnarmorling** [comment]
LGTM, thanks, @jpechane!

**Pull request merged**